contracts/Stage1/SEuroOffering.sol

```diff
@@ -20,7 +20,7 @@ contract SEuroOffering is Ownable, Pausable, Drainable {
     TokenManager public tokenManager;
     BondingCurve public bondingCurve;
 
-    event Swap(string _token, uint256 amountIn, uint256 amountOut);
+    event Swap(address _user, string _token, uint256 amountIn, uint256 amountOut);
```

Comment:
might as well make it `indexed` while we're at it to better filter for addresses :)